types/Promise.d.ts

```diff
@@ -4,6 +4,41 @@
 // Based on roblox-lua-promise v3.1.0
 // https://eryn.io/roblox-lua-promise/
 
+interface PromiseErrorOptions {
+	trace: string;
+	context: string;
+	kind: keyof PromiseErrorConstructor["Kind"];
+}
+
+interface PromiseError {
+	readonly error: string;
+	readonly trace?: string;
+	readonly context?: string;
+	readonly kind?: keyof PromiseErrorConstructor["Kind"];
+	readonly parent?: PromiseError;
+	readonly createdTick: number;
+	readonly createdTrace: string;
+
+	extend(options?: Partial<PromiseErrorOptions>): PromiseError;
+
+	getErrorChain(): Array<PromiseError>;
+}
+
+interface PromiseErrorConstructor {
+	readonly Kind: {
+		readonly ExecutionError: "ExecutionError";
+		readonly AlreadyCancelled: "AlreadyCancelled";
+		readonly NotResolvedInTime: "NotResolvedInTime";
+		readonly TimedOut: "TimedOut";
+	};
+
+	is(anything: any): boolean;
```

Comment:
```suggestion
	is(value: unknown): value is PromiseError;
```

Reply:
Oh that's interesting, I've never seen that syntax before for a return type

Reply:
These are called "type predicates" and are how all of the `@rbxts/t` types work!

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates